automap/src/comm_layer/pcp_pmp_common/linux_specific.rs

```diff
@@ -11,20 +11,31 @@ pub fn linux_find_routers(command: &dyn FindRoutersCommand) -> Result<Vec<IpAddr
         Ok(stdout) => stdout,
         Err(stderr) => return Err(AutomapError::ProtocolError(stderr)),
     };
-    let addresses = output
+    let init: Result<Vec<IpAddr>, AutomapError> = Ok(vec![]);
+    output
         .split('\n')
         .take_while(|line| line.trim_start().starts_with("default "))
-        .map(|line| {
-            let ip_str: String = line
-                .chars()
-                .skip_while(|char| !char.is_numeric())
-                .take_while(|char| !char.is_whitespace())
-                .collect();
-
-            IpAddr::from_str(&ip_str).expect("Bad syntax from ip route")
+        .fold(init, |acc, line| match acc {
+            Ok(mut ip_addr_vec) => {
+                let ip_str: String = line
+                    .chars()
+                    .skip_while(|char| !char.is_numeric())
+                    .take_while(|char| !char.is_whitespace())
+                    .collect();
+
+                match IpAddr::from_str(&ip_str) {
+                    Ok(ip_addr) => {
+                        ip_addr_vec.push(ip_addr);
+                        Ok(ip_addr_vec)
+                    }
+                    Err(e) => Err(AutomapError::FindRouterError(format!(
+                        "Failed to parse an IP from \"ip route\": {:?}",
```

Comment:
It could be that some future version of `ip route` on some future version of the operating system could have an output format that has digits before the real IP address. When that happens, everybody using those versions together is going to start failing here, but it will be really hard to tell why, because we won't be able to see the text you're parsing.

Make this error message show at least the part of the line we think should be the IP address; even better would be to show the entire line, so that whoever is tasked with figuring out what the problem is will have an easier time.